catalog/src/main/kotlin/com/adevinta/spark/catalog/examples/samples/tokens/colors/ColorSample.kt

```diff
@@ -41,71 +43,88 @@ import androidx.compose.ui.tooling.preview.Preview
-        ColorItem(SparkTheme.colors.surface, "surface")
-        ColorItem(SparkTheme.colors.surfaceInverse, "surface inverse")
-        ColorItem(SparkTheme.colors.outline, "outline")
-        ColorItem(SparkTheme.colors.outlineHigh, "outline High")
-        ColorItem(SparkTheme.colors.success, "success")
-        ColorItem(SparkTheme.colors.successContainer, "success Container")
-        ColorItem(SparkTheme.colors.alert, "alert")
-        ColorItem(SparkTheme.colors.alertContainer, "alert Container")
-        ColorItem(SparkTheme.colors.error, "error")
-        ColorItem(SparkTheme.colors.errorContainer, "error Container")
-        ColorItem(SparkTheme.colors.info, "info")
-        ColorItem(SparkTheme.colors.infoContainer, "info Container")
-        ColorItem(SparkTheme.colors.neutral, "neutral")
-        ColorItem(SparkTheme.colors.neutralContainer, "neutral Container")
+        items(tokens) { token ->
+            Row {
+                token.forEach {
+                    ColorItem(it)
+                }
+            }
+        }
     }
 }
 
 @Composable
-private fun RowScope.ColorItem(color: Color, colorName: String) {
+private fun RowScope.ColorItem(color: KProperty0<Color>) {
     Surface(
         modifier = Modifier
             .padding(8.dp)
-            .sizeIn(minWidth = 104.dp, minHeight = 104.dp)
+            .heightIn(104.dp)
             .weight(1f),
+        color = color.get(),
         shape = SparkTheme.shapes.extraLarge,
-        color = color,
         border = BorderStroke(2.dp, SparkTheme.colors.onBackground),
     ) {
         Box(
             modifier = Modifier.padding(8.dp),
             contentAlignment = Alignment.Center,
         ) {
             Text(
-                text = colorName,
+                text = color.name,
                 style = SparkTheme.typography.body2,
                 textAlign = TextAlign.Center,
             )
         }
     }
 }
 
+private val tokensColors
+    @Composable
+    get() = with(SparkTheme.colors) {
+        listOf(
+            ::main,
+            ::mainContainer,
+            ::mainVariant,
+            ::support,
+            ::supportContainer,
+            ::supportVariant,
+            ::accent,
+            ::accentContainer,
+            ::accentVariant,
+            ::basic,
+            ::basicContainer,
+            ::success,
+            ::successContainer,
+            ::alert,
+            ::alertContainer,
+            ::error,
+            ::errorContainer,
+            ::info,
+            ::infoContainer,
+            ::neutral,
+            ::neutralContainer,
+            ::background,
+            ::backgroundVariant,
+            ::surface,
+            ::surfaceInverse,
+            ::outline,
+            ::outlineHigh,
+        ).groupBy {
+            // Group by token name like "main",  "mainContainer" or "mainVariant"
+            it.name.takeWhile { !it.isUpperCase() }
+        }.values.toList()
```

Comment:
nit: I would have kept the group type and renamed the variable to tokensColorsGroups

Reply:
Since LazyColumn items() only take a list as parameter i had to do this :(

Reply:
Yes, but we could do the `toList()` on the LazyColumn call instead?

Reply:
Also, I think we would benefit from moving this val out of this dynamic getter. Isn't it  going to be recomputed everytime?